homeassistant/components/media_player/yamaha.py

```diff
@@ -83,13 +91,40 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         ctrl_url = "http://{}:80/YamahaRemoteControl/ctrl".format(host)
         receivers = rxv.RXV(ctrl_url, name).zone_controllers()
 
+    devices = []
     for receiver in receivers:
-        if receiver.zone not in zone_ignore:
-            hass.data[KNOWN].add(receiver.ctrl_url)
-            add_devices([
-                YamahaDevice(name, receiver, source_ignore,
-                             source_names, zone_names)
-            ], True)
+        if receiver.zone in zone_ignore:
+            continue
+
+        device = YamahaDevice(name, receiver, source_ignore,
+                              source_names, zone_names)
+
+        # Only add device if it's not already added
+        if device.unique_id not in hass.data[DATA_YAMAHA]:
+            hass.data[DATA_YAMAHA][device.unique_id] = device
+            devices.append(device)
+        else:
+            _LOGGER.debug('Ignoring duplicate receiver %s', name)
+
+    def service_handler(service):
+        """Handle for services."""
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+
+        devices = [device for device in hass.data[DATA_YAMAHA].values()
+                   if device.entity_id and device.entity_id in entity_ids]
```

Comment:
Sorry if I wasn't clear. It should be like this:
```py
devices = [
    device for device in hass.data[DATA_YAMAHA].values()
    if not entity_ids or device.entity_id in entity_ids]
```

Reply:
Ah, right, makes more sense. I've pushed a fix now.